tests/test_validation.py

```diff
@@ -665,6 +690,47 @@ def test_wrong_filetype(self):
             target_file, required_fields=['field_a'], excel_ok=False)
         self.assertTrue('could not be opened as a CSV' in error_msg)
 
+    def test_slow_to_open(self):
+        """Test timeout by mocking a CSV that is slow to open"""
+        from natcap.invest import validation
+
+        # make an actual file so that `check_file` will pass
+        path = os.path.join(self.workspace_dir, 'slow.csv')
+        with open(path, 'w') as file:
+            file.write('1,2,3')
+
+        # define a side effect for the mock that will sleep
+        # for longer than the allowed timeout
+        def delay(*args, **kwargs):
+            time.sleep(10)
+
+        # a fake spec with just one key
+        spec = {
+            "mock_csv_path": {
+                "type": "csv",
+                "required": True,
+                "about": "A CSV that will be mocked.",
+                "name": "CSV"
+            }
+        }
+
+        # validate a mocked CSV that will take 10 seconds to return a value
+        args = {"mock_csv_path": path}
+
+        # make a mock function to replace pandas.read_csv
+        # this will sleep for 10 seconds, then return a dataframe
+        mock_read_csv = unittest.mock.Mock(
+            side_effect=delay, 
+            return_value=pandas.DataFrame([['1','2','3']]))
+
+        # replace the copy of pandas.read_csv that's in the validation module
+        # with the mock function, and try to validate
+        with unittest.mock.patch('natcap.invest.validation.pandas.read_csv', mock_read_csv):
```

Comment:
Neat. I haven' done much Mocking. Would that line behave the same as:
```python
with unittest.mock.patch('natcap.invest.validation.check_csv', mock_read_csv)
```
?

Reply:
I think that would work also... to be consistent I'd need to change the `Mock` return value to be a list, not a dataframe. But yes that should also trigger the timeout. Do you think it would be better that way? I guess it decreases the amount of irrelevant code that's run in the test

Reply:
Hmm, I'm not sure about "better" but I'll leave it up to you. Maybe your last comment of decreasing the amount of code that's run makes sense? I guess `pandas.read_csv` is the actual function that would take awhile to load, but to me `check_csv` is a more clear function to mock... I guess 6 in one hand, half dozen in the other.

Reply:
Happy to change it!

Reply:
Turns out that it's not so simple to replace `'natcap.invest.validation.pandas.read_csv'` with `'natcap.invest.validation.check_csv'` because `validate` references `check_csv` via `functools.partial` in the `_VALIDATE_FUNCS` dictionary. Ended up mocking the dictionary instead and replacing `functools.partial(timeout, check_csv)` with `functools.partial(timeout, delay)` :)